src/utility.rs

```diff
@@ -114,163 +200,111 @@ pub(crate) fn get_coverage_perc(covs: &[Value]) -> Result<f64, SifisError> {
+            if cov > 0 {
+                covered_lines += 1.;
+            }
+        }
     }
-    match writer.write_record(&[
-        "TOTAL FILES IGNORED".to_string(),
-        format!("{:?}", files_ignored.len()),
-        "".to_string(),
-        "".to_string(),
-        "".to_string(),
-        "".to_string(),
-    ]) {
-        Ok(_res) => (),
-        Err(_err) => return Err(SifisError::WrintingError()),
-    };
-    match writer.flush() {
-        Ok(_res) => (),
-        Err(_err) => return Err(SifisError::WrintingError()),
-    };
-    Ok(())
+    Ok((covered_lines, tot_lines))
 }
 
 // get the root FuncSpace from a file
 pub(crate) fn get_root(path: &Path) -> Result<FuncSpace, SifisError> {
-    let data = match read_file(path) {
-        Ok(data) => data,
-        Err(_err) => return Err(SifisError::WrongFile(path.display().to_string())),
-    };
-    let lang = match guess_language(&data, path).0 {
-        Some(lang) => lang,
-        None => return Err(SifisError::LanguageError()),
-    };
-    let root = match get_function_spaces(&lang, data, path, None) {
-        Some(root) => root,
-        None => return Err(SifisError::MetricsError()),
-    };
+    let data = read_file(path).map_err(|_| SifisError::WrongFile(path.display().to_string()))?;
+    let lang = guess_language(&data, path)
+        .0
+        .ok_or(SifisError::LanguageError())?;
+    let root = get_function_spaces(&lang, data, path, None).ok_or(SifisError::MetricsError())?;
     Ok(root)
 }
 
-pub(crate) fn get_cumulative_values(metrics: &Vec<Metrics>) -> (Metrics, Metrics, Metrics) {
-    let mut avg = Metrics {
-        sifis_plain: 0.0,
-        sifis_quantized: 0.0,
-        crap: 0.0,
-        skunk: 0.0,
-        file: "AVG".to_string(),
-    };
-    let mut max = Metrics {
-        sifis_plain: 0.0,
-        sifis_quantized: 0.0,
-        crap: 0.0,
-        skunk: 0.0,
-        file: "MAX".to_string(),
-    };
-    let mut min = Metrics {
-        sifis_plain: f64::MAX,
-        sifis_quantized: f64::MAX,
-        crap: f64::MAX,
-        skunk: f64::MAX,
-        file: "MIN".to_string(),
-    };
-    for m in metrics {
+//check complexity of a metric
+// return true if al least one metric exeece a threshold , false otherwise
+pub(crate) fn check_complexity(
+    sifis_plain: f64,
+    sifis_quantized: f64,
+    crap: f64,
+    skunk: f64,
+    thresholds: Vec<f64>,
+) -> bool {
+    sifis_plain > thresholds[0]
+        || sifis_quantized > thresholds[1]
+        || crap > thresholds[2]
+        || skunk > thresholds[3]
+}
+
+// Get AVG MIN MAx and the list of all complex files
+pub(crate) fn get_cumulative_values(
+    metrics: &Vec<Metrics>,
+) -> (Metrics, Metrics, Metrics, Vec<Metrics>) {
+    let mut avg = Metrics::default();
+    let mut min = Metrics::default();
+    let mut max = Metrics::default();
+    avg.file = "AVG".to_string();
+    max.file = "MAX".to_string();
+    min.file = "MIN".to_string();
```

Comment:
Perhaps we can define `avg()`, `max()` and `min()` methods in order to define `avg, min, max` variables